### PR TITLE
[WIP] bpo-32592: Drop support for Windows Vista

### PR DESCRIPTION
**Pull request by vstinner**
Windows Vista extended support ended in April 2017. As stated in PEP
11, Python 3.7 drops support for Windows Vista and now requires
Windows 7 or newer.

https://bugs.python.org/issue32592

### CODE REVIEW

**Comment by vstinner** [comment]
Hum, the change should be mentionned in Doc/whatsnew/3.7.rst.

**Comment by vstinner** [comment]
AppVeyor compilation failed with: "C:\Program Files (x86)\Windows Kits\10\Include\10.0.16299.0\shared\sdkddkver.h(289): fatal error C1189: #error:  NTDDI_VERSION setting conflicts with _WIN32_WINNT setting [C:\projects\cpython\PCbuild\pythoncore.vcxproj]"

File: Include/internal/condvar.h [hunk 1]
```diff
@@ -34,58 +34,13 @@
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 
-/* options */
```
Comment (matrixise) [inline Include/internal/condvar.h]: Maybe you could change the comment before `#define Py_HAVE_CONDVAR` ?

Comment (matrixise) [reply Include/internal/condvar.h]: in fact, should we remove all the references to Vista when we drop the support of Vista ?

**Comment by matrixise** [comment]
@vstinner There is some references to Vista in the `Tools/msi/` directory.

File: Include/internal/condvar.h (hunk 1 above)
Comment (vstinner) [reply Include/internal/condvar.h]: I updated the comment.

**Comment by vstinner** [comment]
> @vstinner There is some references to Vista in the Tools/msi/ directory.

I don't know this directory, so I prefer to leave it unchanged.

**Comment by matrixise** [comment]
@vstinner ok then in this case, the PR seems to be right.

File: Python/condvar.h [hunk 2]
```diff
@@ -97,156 +97,6 @@ PyCOND_TIMEDWAIT(PyCOND_T *cond, PyMUTEX_T *mut, long long us)
...
 
 #elif defined(NT_THREADS)
-/*
- * Windows (XP, 2003 server and later, as well as (hopefully) CE) support
- *
- * Emulated condition variables ones that work with XP and later, plus
- * example native support on VISTA and onwards.
- */
-
-#if _PY_EMULATED_WIN_CV
```
Comment (zooba) [inline Python/condvar.h]: See https://bugs.python.org/issue29871 before pulling this code out - our alternate code isn't 100% correct yet.

Comment (vstinner) [reply Python/condvar.h]: Oh, I can revert this change and propose it in https://bugs.python.org/issue29871.

My change should have no effect on Windows 7 and newer. Why do you prefer to keep the unused emulation code (until bpo-29871 is fixed)?

**Comment by vstinner** [comment]
> AppVeyor compilation failed with: "C:\Program Files (x86)\Windows Kits\10\Include\10.0.16299.0\shared\sdkddkver.h(289): fatal error C1189: #error: NTDDI_VERSION setting conflicts with _WIN32_WINNT setting [C:\projects\cpython\PCbuild\pythoncore.vcxproj]"

@zooba: Do you understand why my change breaks the compilation?

File: Include/internal/condvar.h [hunk 3]
```diff
@@ -22,70 +22,18 @@
...
 /* include windows if it hasn't been done before */
 #define WIN32_LEAN_AND_MEAN
 #include <windows.h>
 
-/* options */
-/* non-emulated condition variables are provided for those that want
- * to target Windows Vista.  Modify this macro to enable them.
- */
-#ifndef _PY_EMULATED_WIN_CV
-#define _PY_EMULATED_WIN_CV 1  /* use emulated condition variables */
```
Comment (eryksun) [inline Include/internal/condvar.h]: Isn't removing this a mistake? I think the `SRWLOCK` version was tried in another issue that was abandoned.

Comment (vstinner) [reply Include/internal/condvar.h]: Hum, this code seems to sensitive and my change may conflict with https://bugs.python.org/issue29871 so I reverted this change and will open a new dedicated PR.

**Comment by vstinner** [comment]
"posixmodule.obj : error LNK2001: unresolved external symbol _GetMaximumProcessorCount [C:\projects\cpython\PCbuild\pythoncore.vcxproj]"

Oh, my posixmodule.c change doesn't work :-(

Extract of pythoninfo:

```
sys.windowsversion: sys.getwindowsversion(major=10, minor=0, build=14393, platform=2, service_pack='')
sys.winver: 3.7-32
sys.version: 3.7.0a4+ (heads/master-2-g1d7f51e:1d7f51e, Jan 18 2018, 05:42:21) [MSC v.1912 32 bit (Intel)]
```

I understand that AppVeyor runs Windows 10 and Python was built by Visual Studio 2017

https://msdn.microsoft.com/fr-fr/library/windows/desktop/dd405489(v=vs.85).aspx

"Remarks: To compile an application that uses this function, set _WIN32_WINNT >= 0x0601. For more information, see Using the Windows Headers."

Aha. I need to try to understand why changing _WIN32_WINNT failed. But my Windows VM is down :-)

**Comment by vstinner** [comment]
Hum, I'm not sure that my PR is correct anymore. I change its status to WIP, I will rework on it, once I get again access to a Windows VM to fix my PR.

**Comment by matrixise** [comment]
Some news about this PR ?

-- 
Stéphane Wirtel - http://wirtel.be - @matrixise

**Pull request closed without merging**